spec/features/admin/enterprises/business_address_form_spec.rb

```diff
@@ -0,0 +1,143 @@
+      end
+
+      scenario 'Missing city field' do
+        fill_in 'enterprise_business_address_attributes_company', with: 'Company'
+        fill_in 'enterprise_business_address_attributes_address1', with: '35 Ballantyne St'
+        fill_in 'enterprise_business_address_attributes_zipcode', with: '3072'
+        select2_select 'Australia', from: 'enterprise_business_address_attributes_country_id'
+        select2_select 'Victoria', from: 'enterprise_business_address_attributes_state_id'
+        fill_in 'enterprise_business_address_attributes_phone', with: '0123456789'
+
+        click_button "Update"
+        expect(page).to have_content("Business address city can't be blank")
+      end
+
+      scenario 'Missing zipcode field' do
+        fill_in 'enterprise_business_address_attributes_company', with: 'Company'
+        fill_in 'enterprise_business_address_attributes_address1', with: '35 Ballantyne St'
+        fill_in 'enterprise_business_address_attributes_city', with: 'Thornbury'
+        select2_select 'Australia', from: 'enterprise_business_address_attributes_country_id'
+        select2_select 'Victoria', from: 'enterprise_business_address_attributes_state_id'
+        fill_in 'enterprise_business_address_attributes_phone', with: '0123456789'
+
+        click_button "Update"
+        expect(page).to have_content("Business address zipcode can't be blank")
+      end
+
+      scenario 'Missing phone field' do
+        fill_in 'enterprise_business_address_attributes_company', with: 'Company'
+        fill_in 'enterprise_business_address_attributes_address1', with: '35 Ballantyne St'
+        fill_in 'enterprise_business_address_attributes_city', with: 'Thornbury'
+        fill_in 'enterprise_business_address_attributes_zipcode', with: '3072'
+        select2_select 'Australia', from: 'enterprise_business_address_attributes_country_id'
+        select2_select 'Victoria', from: 'enterprise_business_address_attributes_state_id'
+
+        click_button "Update"
+        expect(page).to have_content("Business address phone can't be blank")
+      end
+
+      scenario 'Reset form button clears form' do
+        fill_in 'enterprise_business_address_attributes_company', with: 'Company'
+        fill_in 'enterprise_business_address_attributes_address1', with: '35 Ballantyne St'
+        fill_in 'enterprise_business_address_attributes_city', with: 'Thornbury'
+        fill_in 'enterprise_business_address_attributes_zipcode', with: '3072'
+        select2_select 'Australia', from: 'enterprise_business_address_attributes_country_id'
+        select2_select 'Victoria', from: 'enterprise_business_address_attributes_state_id'
+        fill_in 'enterprise_business_address_attributes_phone', with: '0123456789'
+
+        click_button "Update"
+        
+        go_to_business_details
+
+        click_button "Reset Form"
+        expect(page).to have_content("Enterprise \"#{distributor.name}\" has been successfully updated!")
```

Comment:
So "Reset Form" is performing a destructive action on the server?

I would make that more clear then, like "Remove business address". Otherwise I would think that it resets the form on the page to its original values.